docs/review.md

```diff
@@ -58,6 +58,35 @@ When reviewing code, here are some things we look for and also things we avoid:
 * Assign issues only when in progress to indicate to others what can be picked
   up
 
+## Code Quality
+
+In the past, we have occasionally written different kinds of tests for
+Element and the SDKs, but it hasn't been a consistent focus. Going forward, we'd
+like to change that.
+
+* For new features, code reviewers will expect some form of automated testing to
+  be included by default
+* For bug fixes, regression tests are of course great to have, but we don't want
+  to block fixes on this, so we won't require them at this time
+
+The above policy is not a strict rule, but instead it's meant to be a
+conversation between the author and reviewer. As an author, try to think about
```

Comment:
for labs features, are we requiring (or strongly recommending) tests for each incremental PR or just as a requirement for out-of-labs launch? Concern being that labs features tend to change rapidly during their development, and changing tests all the time sounds overly time consuming. Once the feature is (about to be) stable though, writing tests should not be all that difficult given the incremental reviews.

Existing labs features might have missed the opportunity for tests at launch being a requirement, but new (future) labs features should be testable at each incremental PR, hopefully. Difference might just be that a labs feature would potentially want to introduce tests later if a particular part of the feature is undecided or being live-tested.

Reply:
Hmm, good question. 🙂 It's a spectrum for sure. I think at the moment, we would lean towards prioritising our ability to evolve the app via labs features and thus not require tests to appear at the same time as the initial cut of labs feature work, as long as (for new labs features going forward at least) the feature author understands that they are effectively deferring part of their work (adding tests) until later and tests are expected to appear before the feature can ship.

Reply:
I have folded this bit into the text.